src/encoded/static/components/static-pages/placeholders/MdSortableTable.js

```diff
@@ -0,0 +1,643 @@
+            _.all(defaultColWidths, (dcw) => Utils.isNumeric(dcw))) {
+            return _.map(defaultColWidths, (dcw) => dcw > 0 ? dcw : Utils.DEFAULT_COL_WIDTH);
+        }
+        return Array(columns.length).fill(Utils.DEFAULT_COL_WIDTH);
+    }
+
+    getDefaultSortings(props) {
+        return props.columns.map((column) => {
+            let sorting = "both";
+            if (column.defaultSorting) {
+                const defaultSorting = column.defaultSorting.toLowerCase();
+
+                if (defaultSorting == "desc") {
+                    sorting = "desc";
+                } else if (defaultSorting == "asc") {
+                    sorting = "asc";
+                }
+            }
+            return sorting;
+        });
+    }
+
+    sortData(data, sortings) {
+        let sortedData = this.props.data;
+        for (var i in sortings) {
+            const sorting = sortings[i];
+            const column = this.props.columns[i];
+            const key = this.props.columns[i].key;
+            switch (sorting) {
+                case "desc":
+                    if (column.descSortFunction &&
+                        typeof (column.descSortFunction) == "function") {
+                        sortedData = column.descSortFunction(sortedData, key);
+                    } else {
+                        sortedData = this.descSortData(sortedData, key);
+                    }
+                    break;
+                case "asc":
+                    if (column.ascSortFunction &&
+                        typeof (column.ascSortFunction) == "function") {
+                        sortedData = column.ascSortFunction(sortedData, key);
+                    } else {
+                        sortedData = this.ascSortData(sortedData, key);
+                    }
+                    break;
+            }
+        }
+        return sortedData;
+    }
+
+    ascSortData(data, key) {
+        return this.sortDataByKey(data, key, ((a, b) => {
+            if (Utils.parseFloatable(a) && Utils.parseFloatable(b)) {
+                a = Utils.parseIfFloat(a);
+                b = Utils.parseIfFloat(b);
+            }
+            if (a >= b) {
+                return 1;
+            } else if (a < b) {
+                return -1;
+            }
+        }).bind(this));
+    }
+
+    descSortData(data, key) {
+        return this.sortDataByKey(data, key, ((a, b) => {
+            if (Utils.parseFloatable(a) && Utils.parseFloatable(b)) {
+                a = Utils.parseIfFloat(a);
+                b = Utils.parseIfFloat(b);
+            }
+            if (a <= b) {
+                return 1;
+            } else if (a > b) {
+                return -1;
+            }
+        }).bind(this));
+    }
+
+    sortDataByKey(data, key, fn) {
+        const clone = [...data];
+
+        return clone.sort((a, b) =>
+            fn(a[key], b[key])
+        );
+    }
+
+    onStateChange(index) {
+        const sortings = this.state.sortings.map(((sorting, i) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use callback in setState when referencing the previous state. (react/no-access-state-in-setstate)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4682591)